vdr/didweb/util.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package didweb
+
+import (
+	"errors"
+	"github.com/nuts-foundation/go-did/did"
+	"net/url"
+	"strings"
+)
+
+var errInvalidWebDIDURL = errors.New("URL does not represent a Web DID")
+
+// URLToDID converts a URL back to a Web DID (did:web). It removes /did.json and /.well-known/did.json from the path, if present.
```

Comment:
Add a note on escaping. It requires unescaped urls?

Reply:
When `url.URL` is properly used by the caller, it will always be correct? (so, either use `url.Parse` or set the fields properly)